format/spec.md

```diff
@@ -298,16 +298,137 @@ Iceberg tables must not use field ids greater than 2147483447 (`Integer.MAX_VALU
+| path_2        | 225                | 25                    | 1250                    |
+| path_3        | 0                  | 100                   | 1475                    |
+| path_4        | 125                | 25                    | 1475                    |
+
+In the above example, path_3 can be ignored from the calculation of the next manifest's `first_used_identifier` because
+no rows were added to the table. The value of `first_used_identifier` is computed as the sum of all  `added_rows_count` 
+values from manifests already listed in the manifest list added to the snapshot's `first-used-identifier`.
+
+The new `data_file` entries within these manifests have their  `first-used-identifier` field set to null and on read 
+should have the value computed using inheritance from their manifest's `first_used_identifier` value. Any existing
+`data_file` which is written into the manifest should have its value for `first_used_identifier` copied directly.
+
+The manifest path_1 could have the following entries (imputed values shown in parentheses)
+
+| `file_path` | `record_count` | `first_used_identifier` |
+|-------------|----------------|-------------------------|
+| data_1      | 100            | null (1000)         |
+| data_2      | 75             | 800                     |
+| data_3      | 150            | null (1100)             |
+
+Two newly added files, data_1 and data_3, have `first_used_identifier` set to null. This allows writers to add new
+data files to inherit the values for the total number of added rows from other data_files as well as the snapshot's
+`first_used_identifier`. When reading the `first_used_identifier` column, the value of `first_used_identifier` from the 
+manifest  entry is combined with the sum of `record_count` for all previous `data_file` structs that were added in the 
+manifest.
+
+The existing file, data_2, came from a previous table state (not included in this example) and has it's inherited value
+for `first_used_identifier` written out explicitly into the `data_file` struct since it can no longer be computed in 
+the context of the current manifest-list.
+
+Finally, the new table metadata must update `last-used-identifier` to the new highest possible value. In this example, 
+the last largest value for `first_used_identifier` in the manifest-list plus the `added_rows_count` for that manifest 
+(path_4), 1475 + 125  = 1600.
+
+```json
+{
+  "last-used-identifier": 1600
+}
+```
+
+#### Datafile Propagation
+
+New data files added when `row-lineage` is enabled do not require any modification. The columns for `_row_identifiier`
+and `_last_updated` do not need to be present in the file, but if they are present they must be `null` for all rows in
+the file.
+
+Adding new file to the table where fields `_row_identifier` and `_last_updated` are set to non-null values is forbidden.
+
+When read, any null value for `_row_identifier` is computed by adding the `first_used_identifier` of the parent manifest
```

Comment:
> by adding the `first_used_identifier` of the parent manifest to the row's `_pos` 

Is this correct? I thought it should be `first_used_identifier + sum of rows of data files before this + _pos`?

Reply:
Ah yes, it should have said parent manifest datafile entry. That's a little complicated though so let me see if I can clarify that